File: project-assistant-service/src/main/java/com/timelordtty/projectcalendar/model/Project.java
package com.timelordtty.projectcalendar.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 项目实体类
 * 
 * @author TimeLordTTY
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Project {
    
    private Long id;
    private String name;
    private Integer reviewPeriod; // 评审周期（自然日）
    private LocalDate onlineDate; // 项目上网日期
    private Integer registrationPeriod; // 报名期限（工作日）
    private LocalDate registrationEndDate; // 报名截止日期
    private LocalDate earliestReviewDate; // 最早评审日期
    private LocalDateTime expectedReviewTime; // 开标时间
    private LocalDateTime expertReviewTime; // 专家评审时间
    private String remark; // 项目备注
    private Status status = Status.ACTIVE;
    
    /**
     * 构造一个包含基本信息的项目
     * 
     * @param name 项目名称
     * @param reviewPeriod 评审周期
     * @param onlineDate 上网日期
     * @param registrationPeriod 报名期限
     */
    public Project(String name, Integer reviewPeriod, LocalDate onlineDate, Integer registrationPeriod) {
        this.name = name;
        this.reviewPeriod = reviewPeriod;
        this.onlineDate = onlineDate;
        this.registrationPeriod = registrationPeriod;
    }
} 

File: project-assistant-ui/src/types/project.ts
// 项目状态枚举
export enum ProjectStatus {
  ACTIVE = 'ACTIVE',
  COMPLETED = 'COMPLETED',
  CANCELLED = 'CANCELLED',
  EXPIRED = 'EXPIRED'
}

// 项目类型定义
export interface Project {
  id: number;
  name: string;
  reviewPeriod: number; // 评审周期（自然日）
  onlineDate: string; // 项目上网日期
  registrationPeriod: number; // 报名期限（工作日）
  registrationEndDate: string; // 报名截止日期
  earliestReviewDate: string; // 最早评审日期
  expectedReviewTime: string; // 开标时间
  expertReviewTime: string; // 专家评审时间
  remark: string; // 项目备注
  status: ProjectStatus;
}

// 项目统计信息
export interface ProjectStatistics {
  total: number;
  active: number;
  completed: number;
  cancelled: number;
  expired: number;
}

// 日期事件类型
export interface DateEvent {
  id: number;
  title: string;
  date: string;
  type: 'onlineDate' | 'registrationEndDate' | 'earliestReviewDate' | 'expectedReviewTime' | 'expertReviewTime';
  projectId: number;
  projectName: string;
} 